runtime/src/launch/pbs-aprun/launch-pbs-aprun.c

```diff
@@ -393,10 +397,13 @@ static void genQsubScript(int argc, char *argv[], int numLocales) {
 static void chpl_launch_cleanup(void) {
   if (!debug) {
     if (unlink(expectFilename)) {
-      char msg[FILENAME_MAX + 35];
-      snprintf(msg, FILENAME_MAX + 35, "Error removing temporary file '%s': %s",
+      char* msg=(char *)chpl_mem_allocMany((strlen(expectFilename) + 
+                                            strlen(strerror(errno)) + 36), 
+                        sizeof(char), CHPL_RT_MD_FILENAME, -1, 0);
+      snprintf(msg, FILENAME_MAX + 45, "Error removing temporary file '%s': %s",
```

Comment:
@rchinmay: Sorry that it's taken me awhile to get to this.  I have three main comments today, one important, and two related to making the code more readable or contain fewer magic numbers.  The first is the most important, of course, and the place to focus most.

I may be missing something obvious or having a brain freeze, but shouldn't the second argument to the `snprintf()` relate to the size of the `msg` buffer to ensure that we don't overflow it?  E.g., I would've expected a pattern like:

```c
int msgLen = strlen(expectFilename) + strlen(strerror(errno)) + 36;
char* msg = (char*)chpl_mem_allocMany(msgLen, sizeof(char), ...);
snprintf(msg, msgLen, "...");
```

While here, it might also be nice to tuck the magic number '35' into a variable to at least comment on where it comes from.  E.g.,

```c
int extraChars = 35;  // characters reserved for "Error removing temporary file '': "
```

Though it's imprecise, I could even imagine doing:

```c
const char* format = "Error removing temporary file '%s': %s";
int msgLen = strlen(expectFilename) + strlen(strerror(errno)) + strlen(format);
snprintf(msg, msgLen, format, ...);
```

This is inaccurate in that it's counting the `%s` strings as part of the string length, but I believe this will only overcount (and only by a few characters?), so it seems like it might be worthwhile to avoid the fragility of hardcoding the integer?  (You should definitely double-check my thinking, though, to make sure there isn't a problem I'm not seeing).

Let me know what you think.

Reply:
@bradcray 
>second argument to the snprintf() relate to the size of the msg buffer to ensure that we don't overflow it

Yes. Sorry about my mistake here. I will correct this. I might have missed the 2nd argument in ``snprintf()`` while changing this. 
Also I think using ``char* format`` would be better, because if the error message would be changed in future, then explicitly changing the integer (``extraChars``) would not be required and thus would be more prone to changes in such messages in future. Please tell me if I am wrong anywhere.
Thanks,
R. Chinmay.

Reply:
That matches my thinking, thanks.

Reply:
@bradcray Please review. I have made the changes.